include/swift/SILOptimizer/Utils/Generics.h

```diff
@@ -52,20 +52,78 @@ class ReabstractionInfo {
   /// The first NumResults bits in Conversions refer to indirect out-parameters.
   unsigned NumResults;
 
-  /// The function type after applying the substitutions of the original
-  /// apply site.
+  /// The function type after applying the substitutions used to call the
+  /// specialized function.
   CanSILFunctionType SubstitutedType;
 
   /// The function type after applying the re-abstractions on the
   /// SubstitutedType.
   CanSILFunctionType SpecializedType;
 
+  /// The generic environment to be used by the specialization.
+  GenericEnvironment *SpecializedGenericEnv;
```

Comment:
Do you mean the SpecializedGenericEnv change specifically or almost all of the Generics.h API changes?